lib/beaker/dsl/install_utils/pe_utils.rb

```diff
@@ -361,9 +361,30 @@ def do_install hosts, opts = {}
             pre30master = version_is_less(opts[:pe_ver] || master['pe_ver'], '3.0')
           end
 
+          hosts_agent_only = []
+          hosts_not_agent_only = []
+          agent_only_check_needed = !version_is_less(opts.fetch('pe_ver', '3.8'), '3.99')
+          unless agent_only_check_needed
+            hosts.each do |host|
+              if host['pe_ver'] && !version_is_less(host['pe_ver'], '3.99')
```

Comment:
Since you are already cycling through the hosts here it would be possible to construct the arrays of agent-only/not-agent only hosts here.  Also, should probably go into it's own private method to stop do_install from becoming any more unmanageable.